zebra-rpc/src/methods.rs

```diff
@@ -74,6 +116,46 @@ impl Rpc for RpcImpl {
 
         Ok(response)
     }
+
+    fn get_block(&self, height: Height, _verbosity: u8) -> BoxFuture<Result<GetBlock>> {
+        let mut state = self.state_service.clone();
+
+        async move {
+            let request = zebra_state::Request::Block(zebra_state::HashOrHeight::Height(height));
+            let response = state
+                .ready()
+                .await
+                .map_err(|_| Error {
+                    code: ErrorCode::ServerError(0),
+                    message: "State service is not ready".to_string(),
+                    data: None,
+                })
+                .expect("State service should be always ready")
```

Comment:
Quick Fix: return the error, rather than panicking.

```suggestion
                .map_err(|_| Error {
                    code: ErrorCode::ServerError(0),
                    message: "State service is not ready".to_string(),
                    data: None,
                })?
```

Reply:
Actually I think Janito's suggestion is better.

Reply:
Thanks, i think i forgot to delete the expect line when i did this change at https://github.com/ZcashFoundation/zebra/pull/3707/commits/4357788c33046c88b5cc5d660fd70c5a2486c967

Anyway, i applied this suggestion for this: https://github.com/ZcashFoundation/zebra/pull/3707#pullrequestreview-899551549 which makes the whole thing shorter.